lineage/states/stateCommon.py

```diff
@@ -105,3 +105,44 @@ def basic_censor(cell):
             if not cell.get_sister().isLeafBecauseTerminal():
                 cell.get_sister().left.observed = False
                 cell.get_sister().right.observed = False
+
+
+def negative_LL_atonce(x, uncens_obs, uncens_gammas, cens_obs, cens_gammas):
+    """ uses the negative_LL_atonce and passes the vector of scales and the shared shape parameter. """
+    outt = 0.0
+    for i in range(4):
+        outt += negative_LL_sep(x[1 + i], x[0], uncens_obs[i], uncens_gammas[i], cens_obs[i], cens_gammas[i])
+    return outt
+
+
+def gamma_estimator_atonce(gamma_obs, time_cen, gamas, x0=None):
+    """
+    This is a weighted, closed-form estimator for two parameters
+    of the Gamma distribution for estimating shared shape and separate scale parameters of several drug concentrations at once.
+    """
+    # Handle no observations
+    gammas = [np.ones_like(g) if np.sum(g) == 0.0 else g for g in gamas]
+
+    for i, gamma in enumerate(gammas):
+        assert gamma.shape[0] == gamma_obs[i].shape[0]
+    arg1 = [np.squeeze(gamma_obs[i][time_cen[i] == 1]) for i in range(len(gamma_obs))]
+    arg2 = [np.squeeze(gammas[i][time_cen[i] == 1]) for i in range(len(gamma_obs))]
+    arg3 = [np.squeeze(gamma_obs[i][time_cen[i] == 0]) for i in range(len(gamma_obs))]
+    arg4 = [np.squeeze(gammas[i][time_cen[i] == 0]) for i in range(len(gamma_obs))]
+
+    arrgs = (arg1, arg2, arg3, arg4)
+
+    if x0 is None:
+        x0 = np.array([20.0, 2.0, 3.0, 4.0, 5.0])
+
+    def constr(x):
+        return x[2:5] - x[1:4]
+
+    def constr2(x):
+        return 100.0 - x[0]
+
+    def constr3(x):
+        return 20.0 - x[1:5]
```

Comment:
Is this needed? What was happening when the upper bound was just uniformly 100?

Reply:
Maybe I'm wrong, but I expect the shape parameter to be larger than the scale parameter.
Given that `shape x scale` should indicate the average of phase lengths, I thought we should restrict them a bit more because their multiplication should not be a very large number.

Reply:
Should I bring 800 constraint back?

Reply:
I think just 100 upper bound across the board? Mainly I think it should be consistent with the bounds in the separate estimator.

Reply:
Of course. Will do.